db/migrate/20250610173034_add_voided_invoice_id_to_invoices.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddVoidedInvoiceIdToInvoices < ActiveRecord::Migration[8.0]
+  disable_ddl_transaction!
+
+  def change
+    add_reference :invoices, :voided_invoice, type: :uuid, index: {algorithm: :concurrently}
```

Comment:
You're adding the index because you expect to find the regenerated invoices from the voided invoice, like `voided_invoice.regenerated_invoice`?
If you're not sure yet, you can wait and create the index only if needed as you implement the regeneration 👍